### DISCUSSION TITLE
whitespace detection on text

### DISCUSSION THREAD

**Discussion by fiqihfathor** [PaddlePaddle/PaddleOCR#9635] [Q&A]
Hi everyone. I have question. I'm working on OCR for id card. I use the PaddleOCR library for text detection, text recognition, and key information extraction. But the OCR that has been made has the disadvantage of detecting whitespace in text. I have done fine tuning for text recognition but whitespace cannot be detected. Finally, I did image preprocessing before entering OCR. Whitespace can be detected but worsens the detected characters, only a few images have good results. How do I get image preprocessing that can generally be applied to images or maybe there is another way to detect whitespace? Thank you for the help.

**Comment by Marbyun** [comment]
hi bro are you have done with this? i also want to use paddleocr to do OCR for ID Card, and the result is good but don't have white space..

**Comment by Anant808** [comment]
Related: https://github.com/PaddlePaddle/PaddleOCR/issues/5448
This might help: https://stackoverflow.com/questions/76412029/how-to-improve-paddleocr-performance-it-is-sometimes-not-able-to-detect-space-i

I believe that if existing configurations aren't helping then we might have to fine-tune the model by training the pretrained model on our own dataset (i.e. custom dataset) for the model to adapt to our needs. The same is suggested in the stackoverflow link above.
